enterprise/api/v1/views.py

```diff
@@ -406,14 +441,21 @@ def license_revoke(self, request, *args, **kwargs):
         enrollments_by_course_id = self._enrollments_by_course_for_licensed_user(enterprise_customer_user)
 
         enrollment_api_client = EnrollmentApiClient()
+
+        revocation_results = {}
+        any_failures = False
         for course_overview in get_course_overviews(list(enrollments_by_course_id.keys())):
             enterprise_enrollment = enrollments_by_course_id.get(course_overview.get('id'))
+            course_id = str(course_overview.get('id'))
             try:
                 self._revoke_enrollment(enrollment_api_client, enterprise_enrollment, course_overview)
+                revocation_results[course_id] = 'success'
             except EnrollmentModificationException as exc:
-                return Response(str(exc), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+                revocation_results[course_id] = str(exc)
+                any_failures = True
 
-        return Response(status=status.HTTP_204_NO_CONTENT)
+        status_code = status.HTTP_200_OK if not any_failures else status.HTTP_422_UNPROCESSABLE_ENTITY
+        return Response(revocation_results, status=status_code)
```

Comment:
Overall, I think this approach makes sense and gives all the information about whether a course enrollment was properly modified. I wonder if it's worth surfacing whether a course enrollment was moved to audit vs. unenrolled in this response, beyond a blanket "success"?

Reply:
Yeah, that's a good idea.  I'm trying that out now, so the response data might look like:
```
{'success': true, 'message': 'moved to audit'}
```
or
```
{'success': true, 'message': 'unenrolled'}
```
or
```
{'success': false, 'message': 'Could not update to audit track because: murder hornets.'}
```
What do you think?

Reply:
This response structure makes sense to me, and would be easily parseable by a client! If only "murder hornets" were a real reason for a failure here haha 😉